nequip/model/_eng.py

```diff
@@ -67,12 +72,21 @@ def SimpleIrrepsConfig(config):
     #  exclusive, we just insist that if full
     #  and simple are provided, full must be
     #  consistant with simple)
-    for k, v in update.items():
-        if k in config:
-            assert (
-                config[k] == v
-            ), f"For key {k}, the full irreps options had value `{config[k]}` inconsistant with the value derived from the simple irreps options `{v}`"
-        config[k] = v
+    if len(prefix) == "":
+        for k, v in update.items():
+            if k in config:
+                assert (
+                    config[k] == v
+                ), f"For key {k}, the full irreps options had value `{config[k]}` inconsistant with the value derived from the simple irreps options `{v}`"
+            config[k] = v
+    else:
+        for k, v in update.items():
+            if k in config and config[k] != v:
+                Warning(
+                    f"For key {k}, the simple irreps options {v} is overrode by full irreps `{config[k]}`"
+                )
+            else:
+                config[k] = v
```

Comment:
1. why is this allowed when there is a prefix?
2. what is `Warning`? should be `warnings.warn`

Reply:
was thinking the no prefix condition may contaminate the prefix case. Had just removed this block to avoid bugs.